### PR TITLE
Fallback to smartAPI

### PR DESCRIPTION
**Pull request by mnicpt**
Attempt to fix onApprove capturing and instrument fallback scenario for insights into failure.

### CODE REVIEW

**Comment by mstuart** [comment]
lgtm! @bluepnume @abarco ?

File: src/props/onApprove.js [hunk 1]
```diff
@@ -61,21 +61,25 @@ type ActionOptions = {|
...
-        if (isProcessorDecline) {
-            return restart().then(unresolvedPromise);
-        }
+    if (isProcessorDecline) {
+        return restart().then(unresolvedPromise);
+    }
 
-        throw new Error('Order could not be captured');
-    };
+    throw new Error(`Order could not be captured for action: ${ action }`);
```
Comment (bluepnume) [inline src/props/onApprove.js:79]: Probably this message should not hard-code the word 'capture' -- could we actually just do `throw err;` and re-throw?

File: src/api/order.js [hunk 2]
```diff
@@ -88,45 +88,75 @@ export function getOrder(orderID : string, { facilitatorAccessToken, buyerAccess
...
+        }).catch(() => {
+            return callSmartAPI({
+                accessToken: buyerAccessToken,
+                method:      'post',
+                url:         `${ SMART_API_URI.ORDER }/${ orderID }/capture`,
+                headers:     {
+                    [HEADERS.CLIENT_CONTEXT]: orderID
+                }
+            }).then(smartResponse => {
+                getLogger().info(`lsat_uprade_shadow_success_capture`, { orderID });
```
Comment (bluepnume) [inline src/api/order.js:109]: For this message let's add:

- The correlation id from the REST call
- The correlation id from the Smart call

Then we can easily share samples with the api team.

Comment (mnicpt) [reply src/api/order.js:109]: @bluepnume Do the correlation id's come from the response?  Also, this log only is in response to the smartAPI call.

Comment (mnicpt) [reply src/api/order.js:109]: Not sure where to get the corridor from REST call. I see that we log that already if it errors.

File: src/api/order.js [hunk 3]
```diff
@@ -68,65 +68,112 @@ export function createOrderID(order : OrderCreateRequest, { facilitatorAccessTok
     });
 }
 
+const handleSmartResponse = (response, orderID : string, action : string) => {
+    const { headers } = response;
+    const corrID = headers[HEADERS.PAYPAL_DEBUG_ID];
+    
+    getLogger().info(`lsat_uprade_shadow_success_get_${ action }`, { corrID, orderID });
```
Comment (bluepnume) [inline src/api/order.js:75]: Could we also get the corrID for the failed REST call?

Comment (bluepnume) [reply src/api/order.js:82]: Sorry to nitpick :( but can we get the REST corrid in this specific log too? Just so we don't have to search CAL a bunch for two different logs with the same button session id, we'll have both corr ids in one place to compare.

**Comment by bluepnume** [comment]
One small suggested change, plus let's ramp lsat upgrade to 100, then get this merged 👍

File: src/api/order.js (hunk 3 above)
Comment (mnicpt) [reply src/api/order.js]: No need to be sorry.  Appreciate the nitpick.

**Review by bluepnume — APPROVED** [review]
Gonna make up for beating you in poker by approving this one.

**Pull request merged**